### PR TITLE
master - version changes should not effect old versions structure.

### PR DESCRIPTION
**Pull request by cenkingunlugu**
Closes https://github.com/NovelCOVID/API/issues/488

### CODE REVIEW

**Comment by andredippenaar** [comment]
It looks like the json structure changes. Instead of being indexed by highest number of cases, it is not alphabetical.

**Comment by puf17640** [comment]
I saw your issue and know why you want to do this, but we can't just remove the strict parameter.

**Comment by cenkingunlugu** [comment]
> I saw your issue and know why you want to do this, but we can't just remove the strict parameter.

I left the not-strict possibility. Please re-check.

**Comment by puf17640** [comment]
hm that might be doable, i will talk to @ebwinters about it.

File: routes/api_worldometers.js [hunk 1]
```diff
@@ -38,8 +38,10 @@ router.get('/countries', async (req, res) => {
 router.get('/countries/:query', async (req, res) => {
 	const { query } = req.params;
 	const countries = JSON.parse(await redis.get(keys.countries)).map(fixApostrophe);
+	const isNotStrict = req.query.strict === 'false';
+	const isStrict = !isNotStrict;
```
Comment (puf17640) [inline routes/api_worldometers.js:42]: please change to
`const isStrict = !(req.query.strict === 'false')`

**Comment by puf17640** [comment]
also if you are implementing this on 1 endpoint, you have to do it for all

File: routes/api_worldometers.js (hunk 1 above)
Comment (cenkingunlugu) [reply routes/api_worldometers.js]: done. also for all endpoints using strict queryparam.

Comment (puf17640) [reply routes/api_worldometers.js]: my bad, change to `const isStrict = req.query.strict !== 'false'`

**Comment by EliteDaMyth** [comment]
i personally dont think its necessary at all. you can easily do this on your end.

**Comment by cenkingunlugu** [comment]
> i personally dont think its necessary at all. you can easily do this on your end.

This is so unprofessional and disappointing. You behave like you developed something for the first time.

When you open source a project, lots of embedded software can include your software in their implementations. This means, you can not change it.

You can add a version, and your old version should not change ever, because it is a breaking change. In your new version you can do whatever you want, but in the older version (which you implemented and given a structure to users) if you are changing it, it means you will break lots of users data.

anyways, i will not try to teach you what you are doing. but your approach is not acceptable.

**Comment by ebwinters** [comment]
@cenkingunlugu what breaking change was made to an older version?

**Comment by cenkingunlugu** [comment]
Did you see my issue? On last 3 weeks when i have a call like countries/netherlands it was returning netherlands results. Till yesterday it returns netherlands antilles.

**Comment by ebwinters** [comment]
@cenkingunlugu yes, we are reopening aother pr to fix this along with adding more tests. I see what happened

**Comment by puf17640** [comment]
> Did you see my issue? On last 3 weeks when i have a call like countries/netherlands it was returning netherlands results. Till yesterday it returns netherlands antilles.

i totally understand you here and as you can see above, i also think this needs to be fixed to not have a breaking change in v1, we didn't purposely change that, it just happened in the means of fixing other stuff. i have already filed a pr now that is fixing it just like you did and will add tests for the edge cases soon with ethan! 👍 

i also totally agree with you saying that the behaviour of elite was unprofessional, your issue is relevant and needs to be addressed and other contributors didnt even get time to view your pr before it was closed down.
